File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int,int>up;
        for (int i=0;i<nums.size();i++){
            up[nums[i]]++;
        }
        for (auto it : up){
            if (it.second > nums.size()/2){
                return it.first;
            }
        }
        return -1;
    }
};

File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer, Integer> mp = new HashMap<>();
        int n = nums.length;
        for (int eachNum : nums){
            if (!mp.containsKey(eachNum)){
                mp.put(eachNum,1);
            }
            else{
                mp.put(eachNum,mp.get(eachNum)+1);
            }
            if (mp.get(eachNum) > n/2) return eachNum;
        }
        return 0;
    }
}